### PR TITLE
Transform tables into definition lists

### PR DESCRIPTION
**Pull request by teoli2003**
_Exceptions_, or _Options_, or _Parameters_ are definition lists in the modern structure.

I found a bunch of them not following this rule, so I fixed.

There are a few minor fixes along the way, too.

### CODE REVIEW

File: files/en-us/web/api/element/insertadjacenttext/index.md [hunk 1]
```diff
@@ -45,9 +45,8 @@ Void.
 
 ### Exceptions
 
-| Exception     | Explanation                                         |
-| ------------- | --------------------------------------------------- |
-| `SyntaxError` | The `position` specified is not a recognized value. |
+- `SyntaxError` {{domxref("DOMException")}}
+  - : Thrown if the `position` specified is not a recognized value.
```
Comment (wbamberg) [inline files/en-us/web/api/element/insertadjacenttext/index.md:49]: Not introduced by you, but `position` here should be `where` to match the param name (or the param name should be changed).

```suggestion
  - : Thrown if `where` is not a recognized value.
```

File: files/en-us/web/api/htmlmediaelement/setsinkid/index.md [hunk 2]
```diff
@@ -35,9 +35,12 @@ A {{jsxref("Promise")}} that resolves to {{jsxref("undefined")}}.
 
 ### Exceptions
 
-| Exception                            | Explanation                               |
-| ------------------------------------ | ----------------------------------------- |
-| {{domxref("DOMException")}} | No permission to use the requested device |
+- `NotAllowedError` {{domxref("DOMException")}}
+  - : Returned if there is no permission to use the requested device.
+- `NotFoundError` {{domxref("DOMException")}}
+  - : Returned if the `deviceId` does not match any audo output device.
```
Comment (wbamberg) [inline files/en-us/web/api/htmlmediaelement/setsinkid/index.md:41]:
```suggestion
  - : Returned if the `deviceId` does not match any audio output device.
```

File: files/en-us/web/api/idbkeyrange/bound/index.md [hunk 3]
```diff
@@ -48,32 +48,11 @@ bound(lower, upper, lowerOpen, upperOpen)
...
-          <li>
-            The lower key and upper key match and either of the bounds are open.
-          </li>
-        </ul>
-      </td>
-    </tr>
-  </tbody>
-</table>
+- `DataError` {{domxref("DOMException")}}
+  - : Thrown when one of the following conditions is met:<
```
Comment (wbamberg) [inline files/en-us/web/api/idbkeyrange/bound/index.md:52]:
```suggestion
  - : Thrown when one of the following conditions is met:
```

File: files/en-us/web/api/idbkeyrange/lowerbound/index.md [hunk 4]
```diff
@@ -41,11 +41,8 @@ lowerBound(lower, open)
 
 ### Exceptions
 
-This method may raise a {{domxref("DOMException")}} of the following type:
-
-| Exception   | Description                                     |
-| ----------- | ----------------------------------------------- |
-| `DataError` | The value parameter passed was not a valid key. |
+- `DataError` {{domxref("DOMException")}}
+  - : Thrown if the value parameter passed was not a valid key.
```
Comment (wbamberg) [inline files/en-us/web/api/idbkeyrange/lowerbound/index.md:45]:
```suggestion
  - : Thrown if the `value` parameter was not a valid key.
```

File: files/en-us/web/api/idbkeyrange/only/index.md [hunk 5]
```diff
@@ -27,19 +27,17 @@ only(value)
...
 
 ### Exceptions
 
-This method may raise a {{domxref("DOMException")}} of the following types:
-
-| Exception   | Description                                     |
-| ----------- | ----------------------------------------------- |
-| `DataError` | The value parameter passed was not a valid key. |
+- `DataError` {{domxref("DOMException")}}
+  - : Thrown if the value parameter passed was not a valid key.
```
Comment (wbamberg) [inline files/en-us/web/api/idbkeyrange/only/index.md:40]:
```suggestion
  - : Thrown if the `value` parameter was not a valid key.
```

File: files/en-us/web/api/idbkeyrange/upperbound/index.md [hunk 6]
```diff
@@ -40,11 +40,8 @@ upperBound(bound, open)
 
 ### Exceptions
 
-This method may raise a {{domxref("DOMException")}} of the following type:
-
-| Exception   | Description                                     |
-| ----------- | ----------------------------------------------- |
-| `DataError` | The value parameter passed was not a valid key. |
+- `DataError` {{domxref("DOMException")}}
+  - : Thrown if the value parameter passed was not a valid key.
```
Comment (wbamberg) [inline files/en-us/web/api/idbkeyrange/upperbound/index.md:44]: `value` is not the name of a parameter here.

File: files/en-us/web/api/idbobjectstore/clear/index.md [hunk 7]
```diff
@@ -39,34 +39,10 @@ operation are fired.
...
-      <td><code>TransactionInactiveError</code></td>
-      <td>
-        This {{domxref("IDBObjectStore")}}'s transaction is
-        inactive.<br />
-      </td>
-    </tr>
-  </tbody>
-</table>
+- `ReadOnlyError` {{domxref("DOMException")}}
+  - : Thrown if the transaction associated with this operation is in read-only [mode](/en-US/docs/Web/API/IDBTransaction#mode_constants).
```
Comment (wbamberg) [inline files/en-us/web/api/idbobjectstore/clear/index.md:43]: This links to a thing that is deprecated: https://developer.mozilla.org/en-US/docs/Web/API/IDBTransaction#mode_constants - not sure what to do here?

File: files/en-us/web/api/idbobjectstore/createindex/index.md [hunk 8]
```diff
@@ -38,70 +38,31 @@ createIndex(indexName, keyPath, objectParameters)
...
+    - `unique`
+      - : If true, the index will not allow duplicate values for a single key.
+    - `multiEntry`
+      - : If `true`, the index will add an entry in the index for each
+        array element when the `keyPath` resolves to an array.
+        If `false`, it will add one single entry containing the array.
+    - `locale` {{non-standard_inline}}
+      - : Currently Firefox-only (43+), this allows you to specify a locale for the index.
+        Any sorting operations performed on the data via key ranges will then obey sorting rules of that locale
+        (see [locale-aware sortin](/en-US/docs/Web/API/IndexedDB_API/Using_IndexedDB#locale-aware_sorting)).
```
Comment (wbamberg) [inline files/en-us/web/api/idbobjectstore/createindex/index.md:61]:
```suggestion
        (see [locale-aware sorting](/en-US/docs/Web/API/IndexedDB_API/Using_IndexedDB#locale-aware_sorting)).
```

File: files/en-us/web/api/idbobjectstore/createindex/index.md [hunk 9]
```diff
@@ -38,70 +38,31 @@ createIndex(indexName, keyPath, objectParameters)
...
     empty name.
-- keyPath
+- `keyPath`
   - : The key path for the index to use. Note that it is possible to create an index with
     an empty `keyPath`, and also to pass in a sequence (array) as a
     `keyPath`.
-- objectParameters {{optional_inline}}
+- `objectParameters` {{optional_inline}}
 
   - : An {{domxref("IDBIndexParameters")}} object, which can include the following
```
Comment (wbamberg) [inline files/en-us/web/api/idbobjectstore/createindex/index.md:50]:
```suggestion
  - : An object which can include the following
```

(IDBIndexParameters is only referenced here)

File: files/en-us/web/api/idbobjectstore/createindex/index.md [hunk 10]
```diff
@@ -38,70 +38,31 @@ createIndex(indexName, keyPath, objectParameters)
...
+      - : If `true`, the index will add an entry in the index for each
+        array element when the `keyPath` resolves to an array.
+        If `false`, it will add one single entry containing the array.
+    - `locale` {{non-standard_inline}}
+      - : Currently Firefox-only (43+), this allows you to specify a locale for the index.
+        Any sorting operations performed on the data via key ranges will then obey sorting rules of that locale
+        (see [locale-aware sortin](/en-US/docs/Web/API/IndexedDB_API/Using_IndexedDB#locale-aware_sorting)).
+        You can specify its value in one of three ways:
+        - `string`: A string containing a specific locale code, e.g. `en-US`, or `pl`.
+        - `auto`</code>`: The platform default locale will be used (may be changed by user agent settings.)
```
Comment (wbamberg) [inline files/en-us/web/api/idbobjectstore/createindex/index.md:64]:
```suggestion
        - `auto`: The platform default locale will be used (may be changed by user agent settings).
```

File: files/en-us/web/api/idbrequest/readystate/index.md [hunk 11]
```diff
@@ -17,7 +17,7 @@ browser-compat: api.IDBRequest.readyState
 The **`readyState`** read-only property of the
 {{domxref("IDBRequest")}} interface returns the state of the request.
 
-\>Every request starts in the `pending` state. The state changes to
+Every request starts in the `pending` state. The state changes to
```
Comment (wbamberg) [inline files/en-us/web/api/idbrequest/readystate/index.md:20]: There's another table in this page - out of scope for this PR?

File: files/en-us/web/api/idbtransaction/mode/index.md [hunk 12]
```diff
@@ -27,38 +27,17 @@ read-only, or do you want to write to the object stores?) The default value is
 An {{domxref("IDBTransactionMode")}} object defining the mode for isolating access to
```
Comment (wbamberg) [inline files/en-us/web/api/idbtransaction/mode/index.md:27]:
```suggestion
An object defining the mode for isolating access to
```

(again apparently only appears here?)

File: files/en-us/web/api/idbtransaction/mode/index.md [hunk 13]
```diff
@@ -27,38 +27,17 @@ read-only, or do you want to write to the object stores?) The default value is
...
-</table>
+- `readonly`
+  - : Allows data to be read but not changed.
+- `readwrite`
+  - : Allows reading and writing of data in existing data stores to be changed.
+- `versionchange`
+  - : Allows any operation to be performed, including ones that delete and
+    create object stores and indexes. This mode is for updating the version
+    number of transactions that were started using
+    {{domxref("IDBDatabase.setVersion()")}}. Transactions of
```
Comment (wbamberg) [inline files/en-us/web/api/idbtransaction/mode/index.md:38]: This appears to not exist any more: https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Releases/10#indexeddb and is not documented anywhere on MDN. Still I'm not sure how to rework this bit so it makes sense.

File: files/en-us/web/api/idbversionchangeevent/newversion/index.md [hunk 14]
```diff
@@ -22,8 +22,7 @@ database.
 
 ## Value
 
-A [64-bit
-integer](</en-US/docs/NSPR_API_Reference/Long_Long_(64-bit)_Integers>).
+A {{jsxref("Number")}} containing a 64-bit integer.
```
Comment (wbamberg) [inline files/en-us/web/api/idbversionchangeevent/newversion/index.md:25]: Isn't this one of these places where we say, it's not a wrapper object, but a primitive? So should this just be `number`?

File: files/en-us/web/api/idbversionchangeevent/oldversion/index.md [hunk 15]
```diff
@@ -18,15 +18,13 @@ The **`oldVersion`** read-only property of the
...
-0\.
+When the opened database doesn't exist yet, the value of `oldVersion` is 0.
 
 {{AvailableInWorkers}}
 
 ## Value
 
-A [64-bit
-integer](</en-US/docs/NSPR_API_Reference/Long_Long_(64-bit)_Integers>).
+A {{jsxref("Number")}} containing a 64-bit integer.
```
Comment (wbamberg) [inline files/en-us/web/api/idbversionchangeevent/oldversion/index.md:27]: Same comment here about wrapper objects.

File: files/en-us/web/api/idbversionchangeevent/version/index.md [hunk 16]
```diff
@@ -23,8 +23,7 @@ in a [versionchange](/en-US/docs/Web/API/IDBTransaction#version_change) transact
 
 ## Value
 
-A [64-bit
-integer](</en-US/docs/NSPR_API_Reference/Long_Long_(64-bit)_Integers>).
+A {{jsxref("Number")}} containing a 64-bit integer.
```
Comment (wbamberg) [inline files/en-us/web/api/idbversionchangeevent/version/index.md:26]: And here.

**Review by wbamberg — CHANGES_REQUESTED** [review]
Thanks @teoli2003 ! Love to see these tables disappearing.

File: files/en-us/web/api/idbversionchangeevent/newversion/index.md (hunk 14 above)
Comment (teoli2003) [reply files/en-us/web/api/idbversionchangeevent/newversion/index.md:25]: I may be mistaken, but I've understood "Number value" here: https://webidl.spec.whatwg.org/#es-unsigned-long-long

as the Number object. (The capitalization may have fooled me)

Comment (wbamberg) [reply files/en-us/web/api/idbversionchangeevent/newversion/index.md:25]: I have no idea really and don't have the appetite to test it this morning :) so am happy to keep it like this.

File: files/en-us/web/api/idbobjectstore/clear/index.md (hunk 7 above)
Comment (teoli2003) [reply files/en-us/web/api/idbobjectstore/clear/index.md:47]: The mode constants have been deprecated (that is the capital-case constants have been deprecated, like `READONLY` but not the enums that lead to literals: `"read-only"`. They are listed in IDBTransaction.mode which is also not deprecated.

Linking to this instead.

File: files/en-us/web/api/idbrequest/readystate/index.md (hunk 11 above)
Comment (teoli2003) [reply files/en-us/web/api/idbrequest/readystate/index.md:20]: Removed.

File: files/en-us/web/api/idbfactory/databases/index.md [hunk 17]
```diff
@@ -40,12 +40,8 @@ A promise that resolves either to an error or a list of dictionaries, each with
...
 ### Exceptions
 
-This method may raise a {{domxref("DOMException")}} of the following types:
-
-| Attribute                                | Description                                          |
-| ---------------------------------------- | ---------------------------------------------------- |
-| {{exception("SecurityError")}} | The method is called from an opaque origin.          |
-| Other error                              | Specification does not describe all possible errors. |
+- `SecurityError` {{domxref("DOMException")}}
+  - : Thrown if the method is called from an opaque origin.
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply files/en-us/web/api/idbfactory/databases/index.md]: I've linked to the StackOverflow article that has the most extensive info I found. We may need to incorporate this somewhere on MDN, but this is outside the scope of this PR.

Note that I found the link to the Stack Overflow entry in our own MDN article about the `Origin` HTTP header. That is good.

**Comment by teoli2003** [comment]
I think I have fixed all the review comments. (Thanks for them, @wbamberg )

File: files/en-us/web/api/idbrequest/readystate/index.md [hunk 18]
```diff
@@ -17,21 +17,17 @@ browser-compat: api.IDBRequest.readyState
...
-The {{domxref("IDBRequestReadyState")}} of the request, which takes one of the
-following two values:
-
-| Value     | Meaning                 |
-| --------- | ----------------------- |
-| `pending` | The request is pending. |
-| `done`    | The request is done.    |
+The {{domxref("IDBRequestReadyState")}} of the request,
+which is `pending` if the request is still ongoing,
+or `done`if the request has completed.|
```
Comment (wbamberg) [inline files/en-us/web/api/idbrequest/readystate/index.md:30]:
```suggestion
or `done` if the request has completed.|
```

**Review by wbamberg — APPROVED** [review]
👍 thank you @teoli2003 !

**Pull request merged**